client/src/pages/Donations.jsx

```diff
@@ -1,36 +1,155 @@
 import React, { useState, useEffect, useContext } from "react";
 import axios from "axios";
-import { Organization } from "../components/Organization";
-import { Container, Row, Table} from "react-bootstrap";
+import { Container, Table, Form, Button } from "react-bootstrap";
 import { UserContext } from "../utils/auth";
 
 export const Donations = () => {
-    const [contributions, setContributions] = useState({"Feeding America":"$20", "Ocean Conservacy":"$20","Education Through Music":"$20","Direct Relief":"$20"});
+  const { user } = useContext(UserContext);
+  const [donationData, setDonationData] = useState({ name: "", amount: 0 });
+  const [errors, setErrors] = useState({});
+  const [contributions, setContributions] = useState([]);
 
-    const addContributions = () => {
-        
-    }
+  const refreshDonations = async () => {
+    const res = await axios.get("/api/donations", {
+      params: {
+        userId: user.uid,
+      },
+    });
+    setContributions(res.data);
+  };
+
+  //get all donations that the user currently has
+  useEffect(() => {
+    const getDonations = async () => {
+      const res = await axios.get("/api/donations", {
+        params: {
+          userId: user.uid,
+        },
+      });
+      setContributions(res.data);
+    };
+
+    getDonations();
+  }, [user]);
+
+  // finds form errors
+  const findFormErrors = () => {
+    const { name, amount } = donationData;
+    const newErrors = {};
+    // name errors
+    if (name === "") newErrors.name = "cannot be blank!";
```

Comment:
`name === ""` => `!name`

Since an "" and 0 are falsy values in javascript

Reply:
Sounds good will probably refactor a couple other spots which also use this logic.